### PR TITLE
Setup rollbar tracking

### PR DESCRIPTION
**Pull request by cycomachead**
This will log exceptions to Rollbar so we can track errors.
Rollbar also supports tracking JS exceptions.

### CODE REVIEW

**Comment by cycomachead** [comment]
I like this much better than use the lapis-exceptions package, though I did borrow a method from that package.

**Comment by cycomachead** [comment]
Here's the latest example of an error I created locally:
https://rollbar.com/michael.ball/SnapCloud/items/5/?person_page=0&#traceback

This is all working now.

File: app.lua [hunk 1]
```diff
@@ -126,8 +134,18 @@ function app:handle_404()
 end
 
 function app:handle_error(err, trace)
+    -- self.current_user is not available here.
+    local current_user = package.loaded.Users:find(self.session.username)
```
Comment (cycomachead) [inline app.lua]: I really don't understand why this is necessary. 🤷‍♂️

File: app.lua [hunk 2]
```diff
@@ -126,8 +134,18 @@ function app:handle_404()
...
+    if current_user then
+        user_params = current_user:rollbar_params()
+    else
+        user_params = {}
+    end
     print(err)
     print(trace)
+    rollbar.set_person(user_params)
+    rollbar.set_custom_trace(err .. "\n\n" .. trace)
+    rollbar.report(rollbar.ERR, helpers.normalize_error(err))
```
Comment (cycomachead) [inline app.lua]: There are a few supported levels. We are free to call this wherever, which can be useful if we need to check on some other behavior.

Levels:
```
  CRIT  = 'critical',
	  ERR   = 'error',
	  WARN  = 'warning',
	  INFO  = 'info',
	  DEBUG = 'debug',
```

File: app.lua [hunk 3]
```diff
@@ -126,8 +134,18 @@ function app:handle_404()
...
 function app:handle_error(err, trace)
+    -- self.current_user is not available here.
+    local current_user = package.loaded.Users:find(self.session.username)
+    if current_user then
+        user_params = current_user:rollbar_params()
+    else
+        user_params = {}
+    end
     print(err)
     print(trace)
```
Comment (bromagosa) [inline app.lua]: Did you mean to keep this here? Where will it be printed if the server is running in the background?

Comment (cycomachead) [reply app.lua]: I did mean to leave them (they were already there..). I thought those were the statements that were logging stacktraces to the error.log. but that doesn't seem to be the case... So, I guess I should just remove them.

File: app.lua (hunk 1 above)
Comment (cycomachead) [reply app.lua]: Yeah, I think it has something to do with certain objects from a before_filter not being available? I tried on master and couldn't log `self.current_user.username`

File: app.lua [hunk 4]
```diff
@@ -126,8 +134,18 @@ function app:handle_404()
...
+    rollbar.set_person(user_params)
+    rollbar.set_custom_trace(err .. "\n\n" .. trace)
```
Comment (cycomachead) [inline app.lua]: This looks a little funny, but both variables contain different parts of the trace. And normalize_error method helps, but in the case of DB errors, doesn't really grab the best line, so I want to make sure everything is logged.

File: config.lua [hunk 5]
```diff
@@ -35,7 +39,7 @@ config({'development', 'test'}, {
     site_name = 'dev | Snap Cloud',
     port = os.getenv('PORT') or 8080,
     mail_smtp_port = os.getenv('MAIL_SMTP_PORT') or 1025,
-    dns_resolver = 'localhost',
+    dns_resolver = '8.8.8.8',
```
Comment (cycomachead) [inline config.lua]: localhost was having issues resolving on the Mac, but using the DigitalOcean IP in production should still be fine.

**Pull request merged**